fmralign/alignment_methods.py

```diff
@@ -465,3 +469,265 @@ def fit(self, X, Y):
+        self.n_time_points = None
+        self.labels = None
+        self.alphas = None
+
+        if searchlights is None and parcels is None:
+            raise ValueError("searchlights or parcels must be provided")
+
+        if searchlights is not None and parcels is not None:
+            raise ValueError(
+                "searchlights and parcels cannot be provided at the same time"
+            )
+
+        if searchlights is not None:
+            self.regions = searchlights
+        else:
+            self.regions = parcels
+
+        self.dists = dists
+        self.radius = radius
+        self.tuning = tuning
+
+        self.tuning_data = []
+        self.denoised_signal = []
+        self.decomp_method = decomp_method
+        self.n_components = n_components
+        self.n_jobs = n_jobs
+
+    ################################################################
+    # Computing decomposition
+
+    @staticmethod
+    def _tuning_estimator(shared_response, target):
+        """
+        Estimate the tuning matrix for individualized neural tuning.
+
+        Parameters:
+        ----------
+        shared_response : array-like
+             The shared response matrix of shape (n_timepoints, k)
+             where k is the dimension of the sources latent space.
+        target : array-like
+             The target matrix.
+        latent_dim : int, optional
+             The number of latent dimensions (if PCA is used). Defaults to None.
+
+        Returns:
+        --------
+        array-like: The estimated tuning matrix for the given target.
+
+        """
+        if shared_response.shape[1] != shared_response.shape[0]:
+            return (np.linalg.pinv(shared_response)).dot(target)
+        return np.linalg.inv(shared_response).dot(target)
+
+    @staticmethod
+    def _stimulus_estimator(full_signal, n_subjects, latent_dim=None, scaling=True):
+        """
+        Estimates the stimulus matrix for the Individualized Neural Tuning model.
+
+        Parameters:
+        -----------
+        full_signal : ndarray
+             Concatenated signal for all subjects,
+             of shape (n_timepoints, n_subjects * n_voxels).
+        n_subjects : int
+             The number of subjects.
+        latent_dim : int, optional
+             The number of latent dimensions to use. Defaults to None.
+        scaling : bool, optional
+             Whether to scale the stimulus matrix sources. Defaults to True.
+
+        Returns:
+        --------
+        stimulus : ndarray
+            The stimulus matrix of shape (n_timepoints, n_subjects * n_voxels)
+        """
+        n_timepoints = full_signal.shape[0]
+        if scaling:
+            U = svd_pca(full_signal)
+        else:
+            U, _, _ = safe_svd(full_signal)
+        if latent_dim is not None and latent_dim < n_timepoints:
+            U = U[:, :latent_dim]
+
+        stimulus = np.sqrt(n_subjects) * U
+        stimulus = stimulus.astype(np.float32)
```

Comment:
It looks like this is the first (of 14, by my count) places where we explicitly cast to float32. Is it upcasting to float64 at some point ? Working with nibabel, I'd expect we'd have all float32 arrays to start

Reply:
just to be sure tu have float32 (as we multiply by a potential float64 in sqrt)

Reply:
I'll open an issue to track, thanks !